New modifiers

alexjiao125 opened the pull request:
Added triangle wave, sine wave, and hybrid wave module

alexjiao125 committed:
added sine and saw wave modules

still need some fine tuning

alexjiao125 committed:
Added Multi-pulse module

also changed name of sawtooth wave to triangle wave

alexjiao125 committed:
changed puslse maker

now depends on potentiometer on top

myjeeed commented on Modules/Modifier/SineWave/SineWaveATtiny/SineWaveATtiny.ino:
```diff
@@ -0,0 +1,69 @@
+  
+  analogWrite(output_pin, output_val);
+ 
+  delay(freq);
```
Can you also change this to work with the `Millis()` function similar to the PulseMaker code?

jonfroehlich replied:
Also, can you make sure to comment your code and if you're using constants
(which you are, quite liberally), it's often best to define const ints for
these with variable names or, alternatively, to write comments that
describe the constants when they're being used.

On Wed, Jul 6, 2016 at 10:37 AM, Majeed notifications@github.com wrote:

> In Modules/Modifier/SineWave/SineWaveATtiny/SineWaveATtiny.ino
> https://github.com/myjeeed/MakerWear/pull/60#discussion_r69742549:
> 
> > +}
> > +
> > +void loop() {
> > +
> > -  if(i >= 256){
> > -    i = 0;
> > -  }
> > -  int input_val = cutAndMap(input.filteredAnalogRead(AVERAGE), 50, 975, 0, 1023);
> > -  int amplitude = input_val/4;
> > -  int pot_val = analogRead(pot_pin);
> > -  int freq = cutAndMap(pot_val, 0, 1023, 1, 30);
> > -  int output_val = cutAndMap(pgm_read_byte(&sinTable[i++]), 0, 254, 0, amplitude);
> >   +
> > -  analogWrite(output_pin, output_val);
> >   +
> > -  delay(freq);
> 
> Can you also change this to work with the Millis() function similar to
> the PulseMaker code?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/myjeeed/MakerWear/pull/60/files/86c40b93380ffa13dcce9244e3169dad20f4f49b#r69742549,
> or mute the thread
> https://github.com/notifications/unsubscribe/ABi-9SFh2m53nk05UqUDhgFDKjNk13OZks5qS72rgaJpZM4JFUbt
> .

## 

Jon Froehlich
Assistant Professor
Computer Science
University of Maryland, College Park
http://www.cs.umd.edu/~jonf/
@jonfroehlich https://twitter.com/jonfroehlich - Twitter

alexjiao125 committed:
Deleted stuff and renamed other stuff

Deleted hybrid and triangle wave generators.  Renames pulseMaker and
SineWave to SquareGenerator and SineGenerator

alexjiao125 committed:
Modified SineGenerator

now uses the millis function to modify frequency instead of delays

alexjiao125 committed:
added comments to Sine generator
